CoreWebServer/src/au/gov/asd/tac/constellation/webserver/resources/constellation_client.py

```diff
@@ -564,23 +564,45 @@ def open_graph(self, filename):
 
         return graph['id']
 
-    def new_graph(self, schema_name=None):
-        """Open a new graph using the given schema.
+    def new_graph(self, schema_name=None, graph_name=None):
+        """Open a new graph using the given schema and optional graph name.
 
         The default schema is whatever CONSTELLATION's default schema is.
+        If the graph name is left blank, the default name `analytic graph1` etc. will be used.
 
         :param schema_name: The optional schema of the new graph.
+        :param graph_name: The optional graph name for the new graph.
 
         :returns: The id of the new graph.
         """
 
         params = {}
         if schema_name:
             params['schema_name'] = schema_name
+        if graph_name:
+            params['graph_name'] = graph_name
         graph = self.call_service('new_graph', verb='post', args=params).json()
 
         return graph['id']
 
+    def rename_graph(self, graph_id=None, new_graph_name):
```

Comment:
This line causes an error during testing ...
The order of the parameters needs to be changed so that values with defaults are at the end:
`def rename_graph(self, new_graph_name, graph_id=None):`